bin/prover-client/src/operators/checkpoint.rs

```diff
@@ -86,10 +108,18 @@ impl ProvingOp for CheckpointOperator {
             .await?;
         let l1_batch_id = l1_batch_keys.first().expect("at least one").context();
 
+        // Doing the manual block idx to id transformation. Will be removed once checkpoint_info
+        // include the range interms of block_id.
+        // https://alpenlabs.atlassian.net/browse/STR-756
+        let start_l2_idx = self.get_l2id(checkpoint_info.l2_range.0).await?;
+        let end_l2_idx = self.get_l2id(checkpoint_info.l2_range.1).await?;
+        let l2_range = vec![(start_l2_idx, end_l2_idx)];
+
         let l2_batch_keys = self
             .l2_batch_operator
-            .create_task(checkpoint_info.l2_range, task_tracker.clone(), db)
+            .create_task(l2_range, task_tracker.clone(), db)
             .await?;
+
         let l2_batch_id = l2_batch_keys.first().expect("at least one").context();
```

Comment:
Should be checked ahead of time before we do anything so that we are less likely to have dangling jobs.